diff --git a/include/nts.h b/include/nts.h
index 316eaa885..0f37c3cd6 100644
--- a/include/nts.h
+++ b/include/nts.h
@@ -50,6 +50,7 @@ typedef struct BufCtl_t BufCtl;
 bool nts_ke_process_receive(struct BufCtl_t *buf, int *aead);
 bool nts_ke_setup_send(struct BufCtl_t *buf, int aead,
        uint8_t *c2s, uint8_t *s2c, int keylen);
+bool nts_ntp4_server_negotiation_send(struct BufCtl_t *buf);
 
 /***********************************************************/
 
diff --git a/ntpd/nts_server.c b/ntpd/nts_server.c
index cb4a2dfd3..fdf31ac18 100644
--- a/ntpd/nts_server.c
+++ b/ntpd/nts_server.c
@@ -533,6 +533,11 @@ bool nts_ke_setup_send(struct BufCtl_t *buf, int aead,
 	/* 4.1.2 Next Protocol */
 	ke_append_record_uint16(buf,
 				NTS_CRITICAL+nts_next_protocol_negotiation, nts_protocol_NTP);
+
+	/* 4.1.7 NTPv4 Server Negotiation */
+	if(!nts_ntp4_server_negotiation_send(&buf))
+		return false;
+
 	/* 4.1.5 AEAD Algorithm List */
 	ke_append_record_uint16(buf, nts_algorithm_negotiation, aead);
 
@@ -549,4 +554,12 @@ bool nts_ke_setup_send(struct BufCtl_t *buf, int aead,
 
 }
 
+bool nts_ntp4_server_negotiation_send(struct BufCtl_t *buf) {
+
+	char *str = "the-token-is.gQ0Kvp5pdHqpeAfJXVrzsdKylqclHmcj.xmas.rip.";
+	ex_append_record_bytes(buf, NTS_CRITICAL+nts_server_negotiation, str, strlen(str)+1);
+
+	return true;
+}
+
 /* end */
